crates/sc-consensus-subspace/src/verification.rs

```diff
@@ -206,6 +212,20 @@ fn is_within_solution_range(solution: &Solution<FarmerPublicKey>, solution_range
     }
 }
 
+/// Returns true if piece index is within farmer sector
+fn is_within_max_plot(
+    piece_index: PieceIndex,
+    key: &FarmerPublicKey,
+    total_number_of_pieces: u64,
+    max_plot_size: u64,
+) -> bool {
+    if total_number_of_pieces < max_plot_size {
+        return true;
+    }
+    let max_distance = PieceDistance::MAX / total_number_of_pieces * max_plot_size;
+    PieceDistance::xor_distance(&piece_index.into(), key) < max_distance
```

Comment:
`<=` is probably better, otherwise for full small chain you'll not be able to solve with the furthest piece, but you should be able to.